Updates to the supercell code.

wsmorgan opened the pull request:
Updated the supercell.py function so that it returns the points group size for the lattice, the rmax, and the volume factor for the supercells. Also updated it to make the input cell primitive.

wsmorgan committed:
Updated the supercell.py function so that it returns the points group size for the lattice, the rmax, and the volume factor for the supercells. Also updated it to make the input cell primitive.

glwhart reviewed (changes requested):
Obviously, I want to accept this merge request, but I think you should add some comments first.

For example, it'll be confusing to someone (even us, years from now) why you need temporary variables to pass into the make_primitive routine. 

Explain your justification for the changes in the code. Maybe mention them in the merge request comments as well.

wsmorgan committed:
Fixed a bug that Conrad found in supercell, there were HNFs being returned that had 0 determinant because fortran was rounding some interegs down to zero. The bug was fixed using NINT. Also added comments to the code.

wsmorgan committed:
Removed unused subroutine imports and variable declarations from HNF_profiler.f90.

wsmorgan commented:
I've added several comments throughout the python and the fortran code explaining anything that may not have made sense. Global change log is:

1) Fixed a bug Conrad found where fortran was rounding some of the hnf entries down to zero, this was fixed using NINT.

2) Added more outputs, rmax (the norm of the largest vector for the lattice), pgs (the order of the lattice point group), and dets (the determinant of the HNF), to the code to make determining the metric easier (some of these may go away later and most of them will likely be computed in another location once we determine the correct metric for returning the HNFs). 

3) In the fortran code temporary variables were declared for the atom types and positions, this was necessary because `F90wrap` does not work when there are pointers in the subroutine call and `make_primitive` requires points, and because simply setting the pointers to be equal to the input variables results in a segmentation fault during runtime.
